### PR TITLE
Add enable-aem command

### PR DESCRIPTION
**Pull request by yuezh**
The command will enable AzureEnhancedMonitoring extension for VM

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@yuezh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (yuezha). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -587,6 +588,243 @@ __.extend(VMClient.prototype, {
...
+    //Fetch storage account used by the vm
+    var vmStorageAccounts = {};
+    var saProperties;
+
+    StorageUtil.init(this.cli);
+
+    StorageUtil.startProgress($('Getting storage accounts'));
+    var storageAccountsResult = null;
+    try {
+        storageAccountsResult = serviceClients.storageManagementClient.storageAccounts.list(_);
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: please get index = 2 spaces

File: test/commands/arm/vm/arm.vm.enable-aem-tests.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      suite.teardownTest(done);
+    });
+
+    describe('vm', function() {
+
+      it('create for extension get and set', function(done) {
+        this.timeout(vmTest.timeoutLarge);
+        vmTest.checkImagefile(function() {
+          vmTest.createGroup(groupName, location, suite, function(result) {
+		   /*
```
Comment (yugangw-msft) [inline test/commands/arm/vm/arm.vm.enable-aem-tests.js]: please comment for why you commented them out

**Comment by yugangw-msft** [comment]
@huangpf @sauryadas please review, this is under VM

**Comment by yuezh** [comment]
Fixed the review comments so far

File: test/testlistarm.txt [hunk 3]
```diff
@@ -34,6 +34,7 @@ commands/arm/vm/arm.vm-docker-tests.js
 commands/arm/vm/arm.vm-docker-ssh-tests.js
 commands/arm/vm/arm.vm-create-generate-ssh-tests.js
 commands/arm/vm/arm.vm.extension-tests.js
+commands/arm/vm/arm.vm.enable-aem-tests.js
```
Comment (huangpf) [inline test/testlistarm.txt]: Could you please move the extension tests to another section which is right below the VM tests?

```
...
commands/arm/vm/arm.availset-tests.js
# Extension Tests
commands/arm/vm/arm.vm.enable-aem-tests.js
```

File: lib/commands/arm/vm/vmClient._js [hunk 4]
```diff
@@ -587,6 +588,243 @@ __.extend(VMClient.prototype, {
...
+      keys = serviceClients.storageManagementClient.storageAccounts.listKeys(resourceGroupName, storageAccountName, _);
+    } finally {
+      StorageUtil.endProgress();
+    }
+    saProperties.key = keys.storageAccountKeys.key1;
+   
+    return saProperties;
+  },
+
+  _enableStorageAccountAnalytics: function(saProperties, _){
```
Comment (huangpf) [inline lib/commands/arm/vm/vmClient._js]: Can this function be used by other VM commands? If not, is it feasible to move it to a dedicated AEM extension code file? e.g. ```aemExtensionUtil._js```?

File: lib/commands/arm/vm/vmClient._js [hunk 5]
```diff
@@ -587,6 +588,243 @@ __.extend(VMClient.prototype, {
...
+        var setOperation = StorageUtil.getStorageOperation(client, StorageUtil.OperationType.Blob, 'setServiceProperties');
+        var defaultLoggingPerperties = this._getDefaultLoggingPerperties();
+        StorageUtil.performStorageOperation(setOperation, _, defaultLoggingPerperties);
+      }finally{
+        StorageUtil.endProgress();
+      }
+    }
+  },
+
+  _getDefaultLoggingPerperties: function(){
```
Comment (huangpf) [inline lib/commands/arm/vm/vmClient._js]: same

**Comment by yuezh** [comment]
Fixed review comments from @huangpf

File: test/commands/arm/vm/arm.vm.enable-aem-tests.js [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                result.exitStatus.should.equal(0);
+                done();
+              });
+            //}
+          });
+        });
+      });
+
+      it('Enable diagnostics extension on created VM in a resource group', function(done) {
+        var cmd = util.format('vm enable-diag %s %s -a %s --json', groupName, vmPrefix, storageAccount).split(' ');
```
Comment (huangpf) [inline test/commands/arm/vm/arm.vm.enable-aem-tests.js]: Why does the test enable DA extension first? Is it because AEM depends on DA?

**Comment by huangpf** [comment]
LGTM. Thanks.

**Comment by yuezh** [comment]
Resolved the conflict. Would you please kindly merge the PR?

**Pull request merged**